src/libYARP_dev/src/devices/AnalogWrapper/AnalogWrapper.cpp

```diff
@@ -192,9 +192,11 @@ AnalogWrapper::AnalogWrapper(const std::vector<AnalogPortEntry>& _analogPorts, i
     rosTopicName    = "";
     rosNode         = YARP_NULLPTR;
     rosMsgCounter   = 0;
-
+    _rate           = rate;
     sensorId = "AnalogServer";
     createPorts(_analogPorts, rate);
+    ownDevices      = false;
+    subDeviceOwned  = YARP_NULLPTR;
```

Comment:
This would be better in the initialization list, but if you don't want to change it, perhaps they should probably be with the other assignments, above the `createPorts()` call?